sources/main.cpp

```diff
@@ -209,23 +209,36 @@ int main(int argc, char* argv[]) {
 
     // reset combination was pressed on startup
     if (buttonHandler->resetButtonsPressed()) {
-        // create marker file
-        QFile file("/firstrun");
-        if (file.open(QIODevice::WriteOnly)) {
-            file.close();
-        }
-
-        // restore default config
-        if (QFile::exists("/boot/config.json")) {
-            QFile::remove("/boot/config.json");
+        if (QFile::exists(qEnvironmentVariable(Environment::ENV_YIO_APP_DIR, appPath) + "/config.json") &&
+            QFile::exists("/boot/config.json")) {
+            // create marker file
+            QFile file("/firstrun");
+            if (file.open(QIODevice::WriteOnly)) {
+                file.close();
+
+                // rename existing config
+                QFile::rename("/boot/config.json", "/boot/config.json.old");
```

Comment:
Doesn't work if the backup file already exists!
`If a file with the name newName already exists, rename() returns false (i.e., QFile will not overwrite it).`

- Delete old file first
- At least log error if rename fails